src/App.tsx

```diff
@@ -3,71 +1,65 @@
+import { FC, useState, useMemo, useCallback } from 'react';
+
 import { peopleFromServer } from './data/people';
 
-export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+import { Person } from './types/Person';
 
-  return (
-    <div className="container">
-      <main className="section is-flex is-flex-direction-column">
-        <h1 className="title" data-cy="title">
-          {`${name} (${born} - ${died})`}
-        </h1>
+import { Dropdown } from './components';
 
-        <div className="dropdown is-active">
-          <div className="dropdown-trigger">
-            <input
-              type="text"
-              placeholder="Enter a part of the name"
-              className="input"
-              data-cy="search-input"
-            />
-          </div>
+import './App.scss';
 
-          <div className="dropdown-menu" role="menu" data-cy="suggestions-list">
-            <div className="dropdown-content">
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Haverbeke</p>
-              </div>
+export const App: FC = () => {
+  const [query, setQuery] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
 
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Bernard Haverbeke</p>
-              </div>
+  const filteredPerson = useMemo(
+    () =>
+      peopleFromServer.filter(person =>
+        person.name.toLowerCase().includes(query.trim().toLowerCase()),
+      ),
+    [query],
+  );
 
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Antone Haverbeke</p>
-              </div>
+  const handleSelectPerson = useCallback((person: Person | null) => {
+    setSelectedPerson(person);
+  }, []);
 
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Elisabeth Haverbeke</p>
-              </div>
+  const handleChangeQuery = useCallback((str: string) => {
+    setQuery(str);
```

Comment:
According to the checklist, you should make sure that the filter won't be called if the user entered spaces only. To adhere to this, you should add a condition to check if the input string is not just spaces before setting it to the state with `setQuery`.